shoop/admin/modules/contacts/views/detail.py

```diff
@@ -103,18 +102,12 @@ def get_context_data(self, **kwargs):
         context = super(ContactDetailView, self).get_context_data(**kwargs)
         context["companies"] = []
         if isinstance(self.object, PersonContact):
-            order_q = Q(orderer=self.object) | Q(customer=self.object)
             context["companies"] = sorted(
                 self.object.company_memberships.all(), key=(lambda x: force_text(x))
             )
-        else:
-            order_q = Q(customer=self.object)
-        user = getattr(self.object, "user", None)
-        if user:
-            order_q |= Q(creator=user)
         context["contact_groups"] = sorted(
             self.object.groups.all(), key=(lambda x: force_text(x)))
-        context["orders"] = Order.objects.filter(order_q).order_by("-id")
+        context["orders"] = self.object.get_orders().order_by("-id")
```

Comment:
I really think also this should be just just self.oject.customer_orders. In detail orders-tab orders as a customer and creator should be separated IMO. Since we don't even show customer/orderer/creator information in list.